processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/LimitedBufferHashGrouper.java

```diff
@@ -318,6 +336,8 @@ public void close()
   private CloseableIterator<Entry<KeyType>> makeHeapIterator()
   {
     final int initialHeapSize = offsetHeap.getHeapSize();
+    ByteBufferMinMaxOffsetHeap newHeap = offsetHeap.copy();
```

Comment:
can you add a comment here on why we are creating another copy?

Reply:
Done in next commit